internal/cmd/gocdk/init_test.go

```diff
@@ -149,27 +149,42 @@ func TestInit(t *testing.T) {
 }
 func TestInferModulePath(t *testing.T) {
 	testCases := []struct {
-		name       string
-		testGOPATH func(string) (string, func())
-		wantErr    bool
+		name    string
+		setup   func(string) (string, string, func())
+		wantErr bool
 	}{
 		{
 			"no GOPATH entry",
-			func(dir string) (string, func()) {
-				return "", func() {}
+			func(dir string) (string, string, func()) {
+				return "", dir, func() {}
 			},
 			true,
 		},
 		{
-			"single GOPATH entry",
-			func(dir string) (string, func()) {
-				return "GOPATH=" + dir, func() {}
+			"single GOPATH entry, project GOPATH/src",
+			func(dir string) (string, string, func()) {
+				srcDir := filepath.Join(dir, "src")
+				if err := os.Mkdir(srcDir, 0777); err != nil {
+					t.Error(err)
+				}
+				return "GOPATH=" + dir, srcDir, func() {}
 			},
 			false,
 		},
 		{
-			"multiple GOPATH entries",
-			func(dir string) (string, func()) {
+			"single GOPATH entry, project in GOPATH (no src)",
+			func(dir string) (string, string, func()) {
+				return "GOPATH=" + dir, dir, func() {}
+			},
+			true,
+		},
+		{
+			"multiple GOPATH entries, project in GOPATH/src",
+			func(dir string) (string, string, func()) {
+				srcDir := filepath.Join(dir, "src")
+				if err := os.Mkdir(srcDir, 0777); err != nil {
+					t.Error(err)
+				}
```

Comment:
The second temporary directory could be rolled inside this one, which would obviate the need for a cleanup function.